### ISSUE TITLE
[Resnet18] Parameter replacement usage

### ISSUE THREAD

**Issue by QianqianF** [PINTO0309/onnx2tf#399]
### Issue Type

Documentation Feature Request, Others

### onnx2tf version number

1.13.12

### onnx version number

1.13.1

### onnxruntime version number

1.15.0

### onnxsim (onnx_simplifier) version number

0.4.17

### tensorflow version number

2.13.0rc0

### Download URL for ONNX

https://github.com/PINTO0309/onnx2tf/releases/download/0.0.2/resnet18-v1-7.onnx

### Parameter Replacement JSON

```json
{
    "format_version": 1,
    "operations": [
      {
        "op_name": "resnetv15_pool1_fwd/Mean",
        "param_target": "attributes", 
        "param_name": "axes",
        "values": [1, 2] # also tried [0, 1]
      }
    ]
}
```

### Description

1. Purpose: Research development
2. What: 
I tried running`onnx2tf -i resnet18-v1-7.onnx -cotof` or `onnx2tf -i resnet18-v1-7.onnx -ois data:1,3,224,224` or `onnx2tf -i resnet18-v1-7.onnx -ois data:1,3,224,224 -prf replace.json`. Attached is the original onnx graph for resnet18 and the converted tflite-float32 graph respectively
<img width="844" alt="image" src="https://github.com/PINTO0309/onnx2tf/assets/32441593/c8719d1d-a2db-4573-ae7a-86bdfc6ced80">

<img width="968" alt="image" src="https://github.com/PINTO0309/onnx2tf/assets/32441593/d0d7de41-eae2-4bc8-8628-ffa0e7aa8bd4">

It looks like the problem is caused by the conversion from `GlobalAveragePool` to `ReduceMean` being applied to the wrong axis. 

The output of running `onnx2tf -i resnet18-v1-7.onnx -cotof` is attached [here](https://n.ethz.ch/~qifeng/download/log.txt)

3. How: So I tried resolving it with using `replace.json`. However, it's not very successful and I suspect that it's due to me supplying the wrong `op_name`. I have tried `Mean__34` and `resnetv15_pool1_fwd/Mean`. This might be a really stupid question: where should I look to find the correct `op_name` especially in this case the onnx operation is converted to a few different tf operations?

4. Why: I would really like to figure out how to use the package properly so I can convert my other pytorch models to tflite for deploying on edge devices. 
5. Resources: the example parameter replacement json in documentation

**Comment by PINTO0309** [comment]
```bash
onnx2tf -i resnet18-v1-7.onnx -cotof -b 1
```

```
INFO: onnx_output_name: resnetv15_stage2_relu1_fwd tf_output_name: tf.nn.relu_7/Relu:0 shape: (1, 128, 28, 28) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage2_conv4_fwd tf_output_name: tf.math.add_15/Add:0 shape: (1, 128, 28, 28) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage2__plus1 tf_output_name: tf.math.add_17/Add:0 shape: (1, 128, 28, 28) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage2_activation1 tf_output_name: tf.nn.relu_8/Relu:0 shape: (1, 128, 28, 28) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_conv2_fwd tf_output_name: tf.math.add_18/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_conv0_fwd tf_output_name: tf.math.add_19/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_relu0_fwd tf_output_name: tf.nn.relu_9/Relu:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_conv1_fwd tf_output_name: tf.math.add_20/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3__plus0 tf_output_name: tf.math.add_22/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_activation0 tf_output_name: tf.nn.relu_10/Relu:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_conv3_fwd tf_output_name: tf.math.add_23/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_relu1_fwd tf_output_name: tf.nn.relu_11/Relu:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_conv4_fwd tf_output_name: tf.math.add_24/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3__plus1 tf_output_name: tf.math.add_26/Add:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage3_activation1 tf_output_name: tf.nn.relu_12/Relu:0 shape: (1, 256, 14, 14) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_conv2_fwd tf_output_name: tf.math.add_27/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_conv0_fwd tf_output_name: tf.math.add_28/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_relu0_fwd tf_output_name: tf.nn.relu_13/Relu:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_conv1_fwd tf_output_name: tf.math.add_29/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4__plus0 tf_output_name: tf.math.add_31/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_activation0 tf_output_name: tf.nn.relu_14/Relu:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_conv3_fwd tf_output_name: tf.math.add_32/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_relu1_fwd tf_output_name: tf.nn.relu_15/Relu:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_conv4_fwd tf_output_name: tf.math.add_33/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4__plus1 tf_output_name: tf.math.add_35/Add:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_stage4_activation1 tf_output_name: tf.nn.relu_16/Relu:0 shape: (1, 512, 7, 7) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_pool1_fwd tf_output_name: tf.math.reduce_mean/Mean:0 shape: (1, 512, 1, 1) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: flatten_170 tf_output_name: tf.reshape/Reshape:0 shape: (1, 512) dtype: float32 validate_result:  Matches 
INFO: onnx_output_name: resnetv15_dense0_fwd tf_output_name: Placeholder:0 shape: (1, 1000) dtype: float32 validate_result:  Matches 
```

**Comment by QianqianF** [comment]
Thanks for the fast reply. Somehow I didn't try just fixing the batch size and it works completely fine now. 

Nonetheless, if later for whatever reason I would like to perform parameter replacement, could you point me to where to find the op name?

Thanks a lot!

**Comment by PINTO0309** [comment]
It is a problem specific to your environment.
```
pip show onnx onnxruntime onnxruntime-gpu onnxsim tensorflow
```
```
$ docker login ghcr.io
Username (xxxx): {Enter}
Password: {Personal Access Token}
Login Succeeded

$ docker run --rm -it \
-v `pwd`:/workdir \
-w /workdir \
ghcr.io/pinto0309/onnx2tf:1.13.12
```

**Comment by QianqianF** [comment]
Thanks a lot! I'll look into it.

**Comment by QianqianF** [comment]
Update: problem resolved after moving from osx to linux system (in case anyone is having the same issue).

**Comment by PINTO0309** [comment]
Seems to be a bug in onnxruntime for OSX or TensorFlow.

**Comment by QianqianF** [comment]
Ah I figured out what the problem was. I changed `~/.keras/keras.json` to be `"image_data_format": "channels_first"` when trying out a different converter. After changing it back to channels_last, everything is working again on OSX.
